### ISSUE TITLE
'onSelect' return undefined or incorrect element when using search module

### ISSUE THREAD

**Issue by fuhaiq** [Semantic-Org/Semantic-UI#2059]
Hi,

I am using search module for tag suggestion, here's html code
````html
<div class="ui category tag search">
   <div class="ui icon input">
      <input class="prompt" type="text" placeholder="search tags...">
      <i class="search icon"></i>
   </div>
   <div class="results"></div>
</div>
````

and this is js code
````javascript
$.fn.api.settings.api = {
        'tag search': '/tag/search?keywords={query}'
    };
$('.ui.tag.search').search({
        apiSettings: {
            action: 'tag search'
        },
        type: 'category',
        onSelect: function(result, response) {
            console.log(result)
            return false;
        }
    })
````
This is json returned from server.
````javascript
{
  "success": true,
  "results": {
    "category1": {
      "name": "Category 1",
      "results": [
        {
          "title": "S1",
          "description": "Optional Description1"
        },
        {
          "title": "S2",
          "description": "Optional Description2"
        },
        {
          "title": "S3",
          "description": "Optional Description3"
        }
      ]
    },
    "category2": {
      "name": "Category 2",
      "results": [
        {
          "title": "S10",
          "description": "Optional Description10"
        }
      ]
    }
  }
}
````
All the results displayed correctly, I want do some js hack when clicking the element, so I use `onSelect` callback.
'S10' is logged correctly, but 'S2' and 'S3' returned 'undefined' in console after click
![2](https://cloud.githubusercontent.com/assets/5200755/6992252/b53701d8-daf5-11e4-92ac-20fd2648c509.png)

More weird is when I click 'S1', console shows 'S10' element
![4](https://cloud.githubusercontent.com/assets/5200755/6992251/aa2eb34e-daf5-11e4-9bb4-dd766d884a60.png)

Please take a look, thanks!

**Comment by jlukic** [comment]
There were fixed related to this in one of the last patch releases. I'm on my phone but check closed issues.

Are you using latest?

**Comment by fuhaiq** [comment]
@jlukic Yes. I am using 1.11.6
![5](https://cloud.githubusercontent.com/assets/5200755/6995567/26efd3e4-db83-11e4-95da-c0af0a09a2af.png)

**Comment by jlukic** [comment]
Thanks for the clarification. I'll take a look

**Comment by jlukic** [comment]
It looks like search results are removed from dom before click bubbles to dimmer causing [this check](https://github.com/Semantic-Org/Semantic-UI/blob/master/src/definitions/modules/modal.js#L224) to fail.

**Comment by jlukic** [comment]
Investigating this now, thanks for your patience.

**Comment by jlukic** [comment]
I've reworked the offending method, it was a pretty poor implementation previously, and some of the variable names were a bit confusing. 

I've verified against your example using the new $.api({ mockResponse: 'foo'}); syntax which I've also added as a new feature.

**Comment by jlukic** [comment]
Available with `2.0`

**Comment by fuhaiq** [comment]
Cheers!
